server/src/main/java/org/folio/metastorage/server/Storage.java

```diff
@@ -387,12 +387,13 @@ Future<Void> mergeClusters(SqlConnection conn, UUID clusterId, Iterator<UUID> it
   public Future<Void> updateGlobalRecords(LargeJsonReadStream request) {
     return pool.withConnection(conn ->
         getAvailableMatchConfigs(conn).compose(matchKeyConfigs ->
-          new ReadStreamConsumer<JsonObject, Void>()
-          .consume(request, r -> 
-            upsertGlobalRecord(
-              UUID.fromString(request.topLevelObject().getString("sourceId")), 
-              r, matchKeyConfigs))
-        ));
+          conn.close().compose(v ->
+            new ReadStreamConsumer<JsonObject, Void>()
+              .consume(request, r -> 
+                upsertGlobalRecord(
+                    UUID.fromString(request.topLevelObject().getString("sourceId")), 
+                    r, matchKeyConfigs))
+        )));
   }
```

Comment:
If you are not going to use the pool.query in PR #16  then it's simpler to just group differently.. The `withConnection`calls the handler and closes upon completion, but in the original code that's done after the stream handling.. which is wrong.. We want to close it as you do.. Moving two parentheses does it:

```java
    return pool.withConnection(conn ->
            getAvailableMatchConfigs(conn))
        .compose(matchKeyConfigs ->
            new ReadStreamConsumer<JsonObject, Void>()
                .consume(request, r ->
                    upsertGlobalRecord(
                        UUID.fromString(request.topLevelObject().getString("sourceId")),
                        r, matchKeyConfigs))
        );
``